task_stages.go

```diff
@@ -33,19 +37,51 @@ const (
 	PreApply Stage = "pre_apply"
 )
 
+// TaskStageStatus is an enum that represents all possible statuses for a task stage
+type TaskStageStatus string
+
+const (
+	TaskStagePending          TaskStageStatus = "pending"
+	TaskStageRunning          TaskStageStatus = "running"
+	TaskStagePassed           TaskStageStatus = "passed"
+	TaskStageFailed           TaskStageStatus = "failed"
+	TaskStageAwaitingOverride TaskStageStatus = "awaiting_override"
+)
+
+// Permissions represents the permission types for overridding a task stage
+type Permissions struct {
+	CanOverridePolicy *bool `jsonapi:"attr,can-override-policy"`
+	CanOverrideTasks  *bool `jsonapi:"attr,can-override-tasks"`
+	CanOverride       *bool `jsonapi:"attr,can-override"`
+}
+
+// Actions represents a task stage actions
+type Actions struct {
+	IsOverridable *bool `jsonapi:"attr, overridable"`
```

Comment:
I don't think this space is a problem but if you are already committing anything else, maybe just delete it so my eye doesn't twitch

```suggestion
	IsOverridable *bool `jsonapi:"attr,overridable"`
```

Reply:
Looking at the serializer, it looks like the property is "is-overridable"

can you double check that?

Reply:
Good catch :) Fixed.